Forge/src/main/java/vazkii/botania/forge/xplat/ForgeXplatImpl.java

```diff
@@ -300,19 +303,27 @@ public boolean insertFluidIntoPlayerItem(Player player, InteractionHand hand, Fl
 
 	@Override
 	public boolean hasInventory(Level level, BlockPos pos, Direction sideOfPos) {
+		var state = level.getBlockState(pos);
 		var be = level.getBlockEntity(pos);
-		return be != null
-				&& be.getCapability(ForgeCapabilities.ITEM_HANDLER, sideOfPos).isPresent();
+		return be != null && be.getCapability(ForgeCapabilities.ITEM_HANDLER, sideOfPos).isPresent()
+				|| state.getBlock() instanceof WorldlyContainerHolder wch
+						&& wch.getContainer(state, level, pos).getSlotsForFace(sideOfPos).length > 0;
 	}
 
 	@Override
 	public ItemStack insertToInventory(Level level, BlockPos pos, Direction sideOfPos, ItemStack toInsert, boolean simulate) {
 		var be = level.getBlockEntity(pos);
-		if (be == null) {
-			return toInsert;
+		LazyOptional<IItemHandler> cap = LazyOptional.empty();
+		if (be != null) {
+			cap = be.getCapability(ForgeCapabilities.ITEM_HANDLER, sideOfPos);
+		} else {
+			// check for vanilla interface, e.g. composter
+			var state = level.getBlockState(pos);
+			if (state.getBlock() instanceof WorldlyContainerHolder wch) {
+				cap = SidedInvWrapper.create(wch.getContainer(state, level, pos), sideOfPos)[0].cast();
```

Comment:
just call the constructor directly? `create` with only one side just constructs an array needlessly

Reply:
Good point. And technically it's even two arrays, one for the second parameter of the `create` call and another internally for the return value. Fixing with:
```suggestion
				cap = LazyOptional.of(() -> new SidedInvWrapper(wch.getContainer(state, level, pos), sideOfPos));
```